### ISSUE TITLE
[Feature Request] Magisk Hide whitelist

### ISSUE THREAD

**Issue by Douile** [topjohnwu/Magisk#762]
# Suggestion
I think it would be cool if the user to change the mode for setting magisk hide status. Atm there is a blacklist where you select all the apps you want to use magisk hide on. I would like to be able to swap this so instead of selecting apps you want to use magisk hide on, you select apps that will not be blocked by magisk hide.
This would be useful for preventing su requests from any apps not whitelisted, instead of the user having to go through and enable magisk hide for every app, which is annoying when you have a lot of apps.

# Alternatives
You could just add a button at the top allowing user to toggle checkbox on all currently visible apps in the magisk hide menu

# Considerations
Turning Magisk hide on for all app increases cpu usage maybe, idc haven't really looked through magisk hide's source. However I still think it would be nice to have the option to whitelist / turn it on for all apps.

**Comment by Didgeridoohan** [comment]
This again?

https://twitter.com/topjohnwu/status/1039528958289760256

And from the [Magisk documentation](https://topjohnwu.github.io/Magisk/tutorials.html#best-practices-for-magiskhide):
> Add your target app to the MagiskHide list: go to the “MagiskHide” section in Magisk Manager, and check the app you want to hide. Do not abuse MagiskHide! I’ve seen some paranoid people adding a lot of unnecessary apps to the list. This behavior is very likely to not only break MagiskHide, but also Magisk itself or even the whole system (magiskd/magisklogd crashes, system bogged down). Remember: with the current MagiskHide implementation, each additional hide target comes with a large cost; only enable those you are SURE you needed!

**Comment by Douile** [comment]
Ah ok
